cmd/daemon/api_server.go

```diff
@@ -68,11 +75,19 @@ const (
 	ApiEventTypeShuffleContext ApiEventType = "shuffle_context"
 )
 
+type WebAPI struct {
+	Method string
+	Path   string
+	Query  url.Values
+}
+
 type ApiRequest struct {
 	Type ApiRequestType
 	Data any
 
 	resp chan apiResponse
+
+	WebAPI WebAPI
```

Comment:
Instead of adding a separate field to the struct, use the `Data` field like the other ones.